### ISSUE TITLE
Memory issue: 2gb of RAM

### ISSUE THREAD

**Issue by Leftyx** [nopSolutions/nopCommerce#3402]
nopCommerce version: 4.1

Steps to reproduce the problem: create 300 categories and more than 100k products with 3 languages.

Our shop is quite big in terms of categories and products. 
We have 3 different languages.

Every time I switch from a language to another it takes roughly 10 seconds to load the new page.
I did a bit of investigation and it seems that there might be a problem with the release of resources.
I've tried to offload some memory usage to redis but it still goes up and up.
I would expect the memory consumption would be much lower with Redis.

![nopcommerce-memory](https://user-images.githubusercontent.com/171024/47956837-23516400-dfa3-11e8-89cd-f5a23055bef4.jpg)

As you can see from the pictures the average memory for one shop is almost 2 gigabyte of RAM.
When I change the homepage choosing another language the CPU goes up to 30% for 10/15 seconds.
Memory is not released.
At some point it goes down to 1.3gb but never below that threshold.

I've noticed that the Localization Service is called for every piece of information, all the time. I know there's some cache involved but still it requesting one property a time. 
Maybe some grouping ?
Also I think it could be useful to introduce some more performant and compressed format like [mempack](https://github.com/neuecc/MessagePack-CSharp).

I could work on it if there is some interest.

**Comment by hitaspdotnet** [comment]
when you try to switch between languages the log system start creating log

**Comment by Leftyx** [comment]
I've sort of managed to alleviate some of the problems with memory and restarts.
Following some advice I've found in the forum I've switched the Application Pool to run 32bit: Switch on: `Enable 32bit applications`

I've changed the the configuration in `Nop.Web.runtimeconfig.json`:

```
{
  "runtimeOptions": {
    "tfm": "netcoreapp2.1",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "2.1.0"
    },
    "configProperties": {
      "System.GC.Concurrent": false,
      "System.GC.Server": false
    }
  }
}
```
disabling the garbage collection.

and I've changed the configuration in `web.config` running dotnet 32bit:

`<aspNetCore requestTimeout="23:00:00" processPath="C:\Program Files (x86)\dotnet\dotnet.exe" arguments=".\Nop.Web.dll" forwardWindowsAuthToken="false" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" startupTimeLimit="3600" />`

Notice: `processPath="C:\Program Files (x86)\dotnet\dotnet.exe" `

Now the average memory usage has gone done to 300Mb. 
I still see those spikes in CPU usage when requesting a new page but that is related  to some poorly written queries to load localized properties and resources causing a SELECT + 1.
